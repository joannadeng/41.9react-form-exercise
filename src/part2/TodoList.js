import React, {useState} from "react";
import Todo from './Todo';
import NewTodoForm from "./NewTodoForm";
import {v4 as uuid} from 'uuid';

const TodoList = () => {
    const initialState = [
        {id:uuid(), task:"grocery shopping"},
        {id:uuid(), task:"clean house"}
    ]

    const [todoList, setTodoList] = useState(initialState);
    const addTask = (newTask) => {
        setTodoList(todoList => [...todoList,{id:uuid(), ...newTask}])
    }
    console.log(todoList)

    return (
        <div className="TodoList">
            <h3>Chore list:</h3>
            <NewTodoForm addTask={addTask}/>
            <div>
                {todoList.map(({task},idx) => <Todo key={idx} id={idx} task={task}/>)}
            </div>
            
        </div>
    )
}

export default TodoList