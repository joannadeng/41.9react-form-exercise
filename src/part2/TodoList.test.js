import {render,fireEvent} from '@testing-library/react';
import TodoList from './TodoList';

it ('it should render without crashing', () => {
    render(<TodoList />)
})

it ("should match snapshot", () => {
    const {asFragment} = render(<TodoList />);
    expect(asFragment()).toMatchSnapshot()
})

it("should add new todo", function() {
    const {queryByText, getByLabelText} = render(<TodoList />);
    const input = getByLabelText("Task");
    const btn = queryByText("Add!");
    expect(queryByText('3:do chore')).not.toBeInTheDocument();
    fireEvent.change(input,{target:{value:'do chore'}});
    fireEvent.click(btn);
    expect(queryByText('3:do chore')).toBeInTheDocument();
})