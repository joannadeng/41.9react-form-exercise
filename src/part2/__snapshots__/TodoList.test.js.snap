// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should match snapshot 1`] = `
<DocumentFragment>
  <div
    class="TodoList"
  >
    <h3>
      Chore list:
    </h3>
    <div
      class="NewTodoForm"
    >
      <form>
        <label
          for="task"
        >
          Task
        </label>
        <input
          id="task"
          name="task"
          placeholder="task"
          type="text"
          value=""
        />
        <button>
          Add!
        </button>
      </form>
    </div>
    <div>
      <div>
        <p>
          1:grocery shopping
        </p>
      </div>
      <div>
        <p>
          2:clean house
        </p>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
